Example/Example Swift/MultipageDocumentService.swift

```diff
@@ -15,16 +15,28 @@ final class MultipageDocumentsService: DocumentServiceProtocol {
     var partialDocuments: [String: PartialDocumentInfo] = [:]
     var compositeDocument: GINIDocument?
     var analysisCancellationToken: BFCancellationTokenSource?
-    
+    var pendingAnalysisHandler: AnalysisCompletion?
+
     init(sdk: GiniSDK) {
         self.giniSDK = sdk
     }
     
     func startAnalysis(completion: @escaping AnalysisCompletion) {
         let partialDocumentsInfoSorted = partialDocuments
+            .lazy
             .map { $0.value }
             .sorted()
             .map { $0.info }
+        
+        // When a PDF/QrCode document is imported the analysis screen is shown right away, and therefore the analysis
+        // is triggered. There could be the case where the document hadn't been analyzed when this happens,
+        // that's why a reference to the completion block has to be kept. Once the document is uploaded,
+        // the completion block is called (see below).
```

Comment:
This comment is really confusing. It's 4 lines and still doesn't make things more clear. Can you explain the different steps of uploading and polling for extractions why how the issue might arise?